src/word-search/index.js

```diff
@@ -1,3 +1,102 @@
-function wordSearch(words, word) {}
+function wordSearch(words, word) {
+
+  // A Object that holds all the letters in the given words array
+  let list = {}
+  let checked = []
+
+  //#region - A function that creates key-value-pairs of all the letters in the array
+  // and store them in list object
+  function Flag() {
+    words.forEach((element, rootI) => {
+      element.forEach((item, childI) => {
+        list[rootI.toString() + childI.toString()] = item
+      })
+    });
+  }
+  //#endregion
+
+  let wordCopy
+  match = false
+
+  //#region - Iterating through the graph to check the first match
+  words.forEach((element, rootI) => {
+    element.forEach((item, childI) => {
+
+      //calling the fuction for each iteration to remove null values in the list object
+      Flag()
+
+      // an array that collects serial word mathes
+      checked = []
+
+      //making an array copy of the word to search
+      wordCopy = word.split('')
+      if (item == word[0]) {
+
+        // calling the dfs function
+        crawler(rootI, childI, 0)
+
+        if (checked.join('') == word) {
+          match = true
+          return
+        }
+      }
+    })
+  });
+  //#endregion
+
+  //#region - A dfs function that handles searching adjacent characters that matches with the next letter in the word
+  function crawler(row, col, matchIndex) {
```

Comment:
Move this crawler function out of the wordSearch() function, a function should have a single responsibility, the wordSearch function is getting too complex so many things going on in there

Reply:
Oh, I get your point, didn't think it mattered. I'll fix it right away.

Reply:
That way your code is easily maintainable and reusable also

Reply:
done